everest-cpp/init.sh

```diff
@@ -4,15 +4,30 @@
 ## Copyright 2020 - 2022 Pionix GmbH and Contributors to EVerest
 ##
 echo "Installing required tools, initializing workspace, changing into everest-core/build"
+GIT_PROTOCOL="git"
 if [ -d "everest-dev-environment" ] ; then
     echo "Directory 'everest-dev-environment' already exists, trying to update its contents..."
     (cd everest-dev-environment && git pull)
 else
-    git clone git@github.com:EVerest/everest-dev-environment.git || { echo "Could not clone everest-dev-environment. Did you add your SSH key on GitHub and made it available to ssh-agent?"; return; }
+    GIT_CLONE_OUTPUT=$(git clone git@github.com:EVerest/everest-dev-environment.git 2>&1)
+    GIT_CLONE_RETVAL=$?
```

Comment:
Probably we should use
ssh -T git@github.com
here as the command (this would make the following if case more robust, because we only need to check the return value). See https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection